hr_timesheet_sheet_period/tests/test_hr_timesheet_sheet_period.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2016-17 Eficent Business and IT Consulting Services S.L.
+# Copyright 2016-17 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from odoo.exceptions import ValidationError as UserError
+
+
+class TestHrTimesheetSheetPeriod(common.TransactionCase):
+
+    def setUp(self):
+        super(TestHrTimesheetSheetPeriod, self).setUp()
+        self.timeheet_model = self.env['hr_timesheet.sheet']
+        self.fiscal_year_model = self.env['hr.fiscalyear']
+        self.hr_contract_model = self.env['hr.contract']
+        self.data_range_type = self.env['date.range.type']
+
+        self.today_date = datetime.today().date()
+        self.date_start = datetime.today().strftime('%Y-01-01')
+        self.date_end = datetime.today().strftime('%Y-12-31')
+        self.company = self.env.ref('base.main_company')
+        self.employee = self.env.ref('hr.employee_admin')
+
+        self.type = self.data_range_type.create({
+            'name': 'Fiscal year',
+            'allow_overlap': False
+        })
+
+    def create_fiscal_year(self):
+        vals = {
+            'company_id': self.company.id,
+            'date_start': self.date_start,
+            'date_end': self.date_end,
+            'schedule_pay': 'monthly',
+            'payment_day': '2',
+            'name': 'Test Fiscal Year 2018',
+            'type_id': self.type.id
+        }
+
+        fiscal_year = self.fiscal_year_model.create(vals)
+
+        return fiscal_year
+
+    def create_hr_timesheet_sheet(self):
+        timesheet_sheet = self.timeheet_model.create({
+            'employee_id': self.employee.id
+        })
+        return timesheet_sheet
+
+    def test_defaults(self):
+        self.create_fiscal_year()
+        hr_timesheet = self.create_hr_timesheet_sheet()
+        self.assertEqual(
+            hr_timesheet.date_end.strftime('%Y-%m-%d'),
```

Comment:
same as here and also check other

Reply:
@aheficent is there real need to convert to string?

Reply:
wrong tag, should've been @ageficent